src/vehicle.cpp

```diff
@@ -5502,11 +5640,17 @@ void vehicle_part::properties_from_item( const item &used_item )
     // Transfer fuel from item to tank
     const ammotype &desired_liquid = vpinfo.fuel_type;
     if( used_item.charges > 0 && desired_liquid == fuel_type_battery ) {
-        amount = std::min<int>( used_item.charges, vpinfo.size );
+        item bat( fuel_type_battery, calendar::turn );
+        bat.charges = std::min<int>( used_item.charges, vpinfo.size * 100 );
+        items.push_front( bat );
     } else if( !used_item.contents.empty() ) {
-        const item &liquid = used_item.contents[0];
-        if( liquid.type->id == default_ammo( desired_liquid ) ) {
-            amount = std::min<int>( liquid.charges, vpinfo.size );
+        const item &liquid = used_item.contents.front();
+        if( desired_liquid.empty() || liquid.type->id == desired_liquid ) {
+            int mul = liquid.type->ammo.get() == nullptr ? 1 : liquid.type->ammo->def_charges;
```

Comment:
If you translate the charges here, you *have* to translate them back in `properties_to_item` with the very same factor. Why not just take the liquid item itself, as you do in `properties_to_item`?

But this is not used currently not used, because installing a filled container is not supported by the vehicle interaction system anyway.